pkg/agent/prommetrics.go

```diff
@@ -45,11 +95,71 @@ func makePrometheusParts(globalstate *agentState) (PromMetrics, *prometheus.Regi
 			[]string{"code"},
 		)),
 		informantRequestsInbound: util.RegisterMetric(reg, prometheus.NewCounterVec(
-			prometheus.CounterOpts{Name: "autoscaling_agent_informant_inbound_requests_total",
+			prometheus.CounterOpts{
+				Name: "autoscaling_agent_informant_inbound_requests_total",
 				Help: "Number of HTTP requests from vm-informants received by autoscaler-agents",
 			},
 			[]string{"endpoint", "code"},
 		)),
+		informantRequestedChange: resourceChangePair{
+			cpu: util.RegisterMetric(reg, prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Name: "autoscaling_agent_informant_requested_cpu_change_total",
+					Help: "Total change in CPU requested from the informant(s)",
+				},
+				[]string{"direction"},
+			)),
+			mem: util.RegisterMetric(reg, prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Name: "autoscaling_agent_informant_requested_mem_change_total",
+					Help: "Total change in memory (in MiB) requested from the informant(s)",
+				},
+				[]string{"direction"},
+			)),
+		},
+		informantApprovedChange: resourceChangePair{
+			cpu: util.RegisterMetric(reg, prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Name: "autoscaling_agent_informant_approved_cpu_change_total",
+					Help: "Total change in CPU approved by the informant(s)",
+				},
+				[]string{"direction"},
+			)),
+			mem: util.RegisterMetric(reg, prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Name: "autoscaling_agent_informant_approved_mem_change_total",
+					Help: "Total change in memory (in MiB) approved by the informant(s)",
+				},
+				[]string{"direction"},
+			)),
+		},
+
+		// ---- NEONVM ----
+		neonvmRequestsOutbound: util.RegisterMetric(reg, prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "autoscaling_agent_neonvm_outbound_requests_total",
+				Help: "Number of k8s patch requests to NeonVM objects",
+			},
+			[]string{"result"},
```

Comment:
what's the cardinality of this

Reply:
In theory it could be large, I think? It's roughly the same idea as in #287 — record "ok" or the root cause of an error.

I'm expecting that _realistically_ it should be quite small (see e.g. https://neonprod.grafana.net/d/aa1653e5-0c19-492f-9da3-b6e6336181d0/neon-autoscaling?orgId=1&viewPanel=21&from=1687110644779&to=1687118972276)

---

In any case, I've added a comment to explain that.

Reply:
Yeah I mean, I'd definitely be more certain about the standard library's HTTP errors rather than the K8s API errors, but at the end of the day it boils down to HTTP (ish) anyways, so I'm figuring it's probably fine.

As long as they aren't creating the base error with formatting, then there's a finite number — and I'd expect that we're never actually going to run into very many different types of errors if we do.

IME keeping this kind of thing around in metrics can be particularly useful when you're not sure how many distinct things are going wrong — like in the link above.